src/source/Signaling/LwsApiCalls.c

```diff
@@ -1449,14 +1468,55 @@ STATUS sendLwsMessage(PSignalingClient pSignalingClient, PCHAR pMessageType, PCH
     size = SIZEOF(pSignalingClient->pOngoingCallInfo->sendBuffer) - LWS_PRE;
     CHK(writtenSize <= size, STATUS_SIGNALING_MAX_MESSAGE_LEN_AFTER_ENCODING);
 
+    // Start off with an empty string
+    encodedIceConfig[0] = '\0';
+
+    // In case of an Offer, package the ICE candidates only if we have a set of non-expired ICE configs
+    if (messageType == SIGNALING_MESSAGE_TYPE_OFFER && pSignalingClient->iceConfigCount != 0 && GETTIME() <= pSignalingClient->iceConfigExpiration) {
+        curTime = GETTIME();
+
+        // Start the ice infos by copying the preamble, then the main body and then the ending
+        STRCPY(encodedIceConfig, SIGNALING_ICE_SERVER_LIST_TEMPLATE_START);
+        iceConfigLen = ARRAY_SIZE(SIGNALING_ICE_SERVER_LIST_TEMPLATE_START) - 1; // remove the null terminator
+
+        for (iceCount = 0; iceCount < pSignalingClient->iceConfigCount; iceCount++) {
+            encodedUris[0] = '\0';
+            for (uriCount = 0; uriCount < pSignalingClient->iceConfigs[iceCount].uriCount; uriCount++) {
+                STRCAT(encodedUris, "\"");
+                STRCAT(encodedUris, pSignalingClient->iceConfigs[iceCount].uris[uriCount]);
+                STRCAT(encodedUris, "\",");
+            }
+
+            // remove the last comma
+            urisLen = STRLEN(encodedUris);
+            encodedUris[--urisLen] = '\0';
```

Comment:
If `pSignalingClient->iceConfigs[iceCount].uriCount` is `0` then we will try to access `encodedUris[-1]`

Is it possible for `pSignalingClient->iceConfigs[iceCount].uriCount` to be `0`, should we add validation for this and/or add if/else block around LN1491/1492 to only enter if `uriCount > 0`

Reply:
This would not be the case as this is an internal structure that gets set only after being validated. This said, it's a good idea to validate. Will add a validation earlier in the if clause.